airflow-core/src/airflow/api_fastapi/core_api/routes/public/xcom.py

```diff
@@ -241,6 +246,7 @@ def create_xcom_entry(
             )
 
     # Check existing XCom
+    request_body.key = unquote(request_body.key)
```

Comment:
Setting this back to request.body feels a little bit out of place. Since this is a JSON body with the key in it, we shouldn't have to do any translation or alteration of the key.